pkg/operator/etcdmemberlistercontroller/etcdmemberlistercontroller.go

```diff
@@ -143,3 +147,31 @@ func (c *EtcdMemberListerController) getMembersFromMap(membersMap map[string]etc
 	}
 	return members
 }
+
+func (c *EtcdMemberListerController) GetMembers() []etcdserverpb.Member {
+	members := make([]etcdserverpb.Member, 0, len(c.membersCache.cache))
+	defer c.membersCache.mux.RUnlock()
+
+	c.membersCache.mux.RLock()
```

Comment:
it's not wrong what you wrote here, but for better readability I would put the locking to the start of the method:

```
	c.membersCache.mux.RLock()
	defer c.membersCache.mux.RUnlock()

	members := make([]etcdserverpb.Member, 0, len(c.membersCache.cache))
        ...
```

So when people add more lines later, they have an easier time to reason about the scope of the lock.

Reply:
+1